### PR TITLE
Fix VB using `ISingleValueCaseClause` as `ElseCaseClause`'s corresponding node in IOperation tree

### PR DESCRIPTION
**Pull request by genlu**
Unlike C# using a `BoundSwaitchLabel` with  `ExpressionOpt == null` to represent a `default` case, in VB, the bound node for `Case Else` statement is not a `BoundSimpleCaseClause` with both `ValueOpt` and `ConditionOpt` equal to `Nothing`. It actually uses a `BoundCaseStatement` with an empty `CaseClauses` array.

This is a part of #6894

@JohnHamby, you have handled it correctly in `BoundCaseBlock`, could you please take a look to see if this change makes sense?

### CODE REVIEW

**Comment by JohnHamby** [comment]
So the code that looked like complete nonsense was actually correct? It probably needs a comment to save the next person who looks at it from thinking it's crazy.

The tests are nice. I sign off.

**Comment by genlu** [comment]
OK, I think this is ready for review. Please take a look

@srivatsn @mavasani @AlekseyTs @amcasey

**Comment by genlu** [comment]
@dotnet-bot test windows please

**Comment by genlu** [comment]
@dotnet-bot Test this please

**Comment by genlu** [comment]
@dotnet-bot Test ETA please

File: src/Compilers/VisualBasic/Portable/BoundTree/Statement.vb [hunk 1]
```diff
@@ -86,6 +86,8 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
 
         Private ReadOnly Property IClauses As ImmutableArray(Of ICaseClause) Implements ICase.Clauses
             Get
+                ' `CaseElseClauseSyntax` is bound to `BoundCaseStatement` with an empty list of case clauses, 
+                ' so we explicitly create an IOperation node for Case-Else clause to differentiate it from Case clause.
                 If Me.CaseStatement.CaseClauses.IsEmpty Then
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Portable/BoundTree/Statement.vb]: Can this be true for erroneous Case clause? We should probably make this check stronger, i.e. make sure the syntax is in fact Case Else.

File: src/Compilers/VisualBasic/Test/Semantic/Diagnostics/OperationAnalyzerTests.vb [hunk 2]
```diff
@@ -171,11 +171,19 @@ Class C
...
             Case > 100000
                 Exit Select
+        End Select   
+
+        Select Case x
+            Case Else
+                Exit Select
+        End Select     
+
+        Select Case x
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Test/Semantic/Diagnostics/OperationAnalyzerTests.vb]: Consider adding tests for erroneous Case clauses.

**Comment by AlekseyTs** [comment]
LGTM, modulo the two comments above.

File: src/Compilers/VisualBasic/Portable/BoundTree/Statement.vb [hunk 3]
```diff
@@ -86,7 +86,9 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
 
         Private ReadOnly Property IClauses As ImmutableArray(Of ICaseClause) Implements ICase.Clauses
             Get
-                If Me.CaseStatement.CaseClauses.IsEmpty Then
+                ' `CaseElseClauseSyntax` is bound to `BoundCaseStatement` with an empty list of case clauses, 
+                ' so we explicitly create an IOperation node for Case-Else clause to differentiate it from Case clause.
+                If Me.CaseStatement.CaseClauses.IsEmpty AndAlso Me.CaseStatement.Syntax.Kind() = SyntaxKind.CaseElseStatement Then
```
Comment (genlu) [inline src/Compilers/VisualBasic/Portable/BoundTree/Statement.vb]: @AlekseyTs Is this what you had in mind?

Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Portable/BoundTree/Statement.vb]: @genlu Yes, perfect.

**Comment by genlu** [comment]
@dotnet-bot Test windows please

**Comment by genlu** [comment]
@dotnet-bot Retest this, please.

**Pull request merged**